pydy/system.py

```diff
@@ -134,43 +159,6 @@ def __init__(self, eom_method, constants=None, specifieds=None,
         self._evaluate_ode_function = None
 
     @property
-    def coordinates(self):
-        """Returns a list of the symbolic functions of time representing the
-        system's generalized coordinates."""
-        if sympy_equal_to_or_newer_than('0.7.6'):
-            return self.eom_method.q[:]
-        else:
-            return self.eom_method._q
-
-    @property
-    def speeds(self):
-        """Returns a list of the symbolic functions of time representing the
-        system's generalized speeds."""
-        if sympy_equal_to_or_newer_than('0.7.6'):
-            return self.eom_method.u[:]
-        else:
-            return self.eom_method._u
-
-    @property
-    def states(self):
-        """Returns a list of the symbolic functions of time representing the
-        system's states, i.e. generalized coordinates plus the generalized
-        speeds. These are in the same order as used in integration (as
-        passed into evaluate_ode_function) and match the order of the mass
-        matrix and forcing vector.
-
-        """
-        return self.coordinates + self.speeds
-
-    @property
-    def eom_method(self):
-        """This is a sympy.physics.mechanics.KanesMethod. The method used to
-        generate the equations of motion. Read-only.
-
-        """
-        return self._eom_method
-
-    @property
```

Comment:
I'm glad I went through this exercise because this is helping highlight where our goals are differing. I'm wanting something that's for more generalized dynamical system and it seems you would prefer to keep it focused on mechanical dynamics? When I look at the bicycle example, I can see where you're coming from; it's quite a sophisticated model and I'm guessing there are significant steps to help with the numerics by taking advantage of mathematical features like the minimal mass matrix form. Then again, it also seems that much of the work is done manually, and would be difficult for System to do even if it had separated out the M and A matrices and the F and G expressions. I didn't see the use of the pydy System in either example.

For most of my applications, I think I'll generally have much more computational power than states, so the difference between the full and minimal mass matrices probably isn't worth worrying about. This general architecture also makes sense to me from a purely development point of view. I would keep all the things your talking about (units, separate differential kinematic equations and dynamic equations, etc) in the purely symbolic *Method side. Then when I'm ready to simulate, I don't need the integrator to care about those things-- I would just choose whether to use the full mass matrix form or the explicit form. 

I think this decoupling would also intuitively support more arbitrary systems like the Van Der Pol oscillator which don't really have those semantics or even controllers/observers where the semantics don't really matter.